utilities/project_helpers/scripts/update-message-xml-translations.py

```diff
@@ -0,0 +1,52 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+
+import argparse
+import os
+import lxml.etree as lxml
+
+from check_message_lib import find_language_file_name, get_xml_keys
+
+arg_parser = argparse.ArgumentParser(description="Add English XML messages missing in the language's messages, marked with @TODO=TRANSLATE.")
+arg_parser.add_argument('-lang', required=True, help='Language (as a 2-letter code) of the messages file')
+arguments = arg_parser.parse_args()
+language = arguments.lang
+
+script_directory = os.path.dirname(os.path.realpath(__file__))
+os.chdir(script_directory)
+
+english_file_name = find_language_file_name('en', 'xml')
+english_keys = get_xml_keys(english_file_name)
+other_file_name = find_language_file_name(language, 'xml')
+other_keys = get_xml_keys(other_file_name)
+
+if (other_keys == english_keys):
+    print('\nThe sets of message keys in {} and {} are already the same.'.format(english_file_name, other_file_name))
+else:
+    current_map = {}
+    parser = lxml.XMLParser(remove_blank_text=True)
+    other_tree = lxml.parse(other_file_name, parser)
+    other_root = other_tree.getroot()
+    for message in other_root:
+        if message.tag is lxml.Comment:
+            other_root.remove(message)
+        else:
+            key = message.get('key')
+            current_map[key] = message
+            other_root.remove(message)
+    english_tree = lxml.parse(english_file_name)
+    english_root = english_tree.getroot()
+    for message in english_root:
+        if (message.tag != lxml.Comment):
+            for element in message.xpath('descendant-or-self::*'):
+                element.tag = element.tag[element.tag.index('}')+1:]
```

Comment:
What do these lines do? Seems that you are looking for `}` in  the tag name?

moreover
```
sources/clarin-dspace$ (gh-merge/1000)> python utilities/project_helpers/scripts/update-message-xml-translations.py sl

Constructing temporary joint xml /tmp/messages-en.xml from all English messages.xml:
  /mnt/c/Users/ko_ok/sources/clarin-dspace/dspace-xmlui/src/main/webapp/i18n/messages.xml
  /mnt/c/Users/ko_ok/sources/clarin-dspace/dspace-xmlui/src/main/resources/aspects/SwordClient/i18n/messages.xml
  /mnt/c/Users/ko_ok/sources/clarin-dspace/dspace-xmlui/src/main/resources/aspects/Discovery/i18n/messages.xml
  /mnt/c/Users/ko_ok/sources/clarin-dspace/dspace-xmlui/src/main/resources/aspects/XMLWorkflow/i18n/messages.xml
Traceback (most recent call last):
  File "utilities/project_helpers/scripts/update-message-xml-translations.py", line 42, in <module>
    element.tag = element.tag[element.tag.index('}')+1:]
ValueError: substring not found
```
But I must say `python utilities/project_helpers/scripts/update-message-xml-translations.py sl` finished on one of four runs, which is even more confusing

Reply:
Yeah, this is hacky, and, as your test runs indicate, not robust. Curly braces are used to wrap a namespace in tag names (e.g., "`{http://apache.org/cocoon/i18n/2.1}message`"). In constructing the temporary joint English messages.xml (which is in check_message_lib.py, already from before), I had to deal with the fact that for some reason, some of the English messages.xml files have a default namespace (xmlns attribute) and others don't. But it looks like the order in which I construct the joint xml isn't deterministic (oops, set instead of list!), and that the namespace in the joint xml is not guaranteed. If and when this joint xml doesn't have a default namespace, your ValueError must be triggered. 

Ok, I can fix the joint xml construction, and maybe it's best to make sure it does *not* have a namespace (since messages_cs.xml and messages_sl.xml don't). And then these lines also become redundant and can be removed.